app/models/listing.rb

```diff
@@ -22,6 +22,11 @@ class Listing < ApplicationRecord
   after_commit :remove_from_elasticsearch, on: [:destroy]
   acts_as_taggable_on :tags
   has_many :credits, as: :purchase, inverse_of: :purchase, dependent: :nullify
+  has_many :listing_endorsements,
+           -> { where(approved: true) },
```

Comment:
why the default scope? should we be explicit here?

`listings.endorsements.approved`

what do you think?

Reply:
okay, I thought of this too, but I do not think it would work, at best it would get only listings that has an endorsement. 

Being explicit here isn't bad, because honestly if an endorsement exist and it is not approved we do not care about it, am I wrong? @rhymes

Reply:
The problem with hidden scopes is that you don't know the other items exist until it's too late usually. 

To make this work you have to add a scope in `ListingEndorsement`, something like:

```ruby
scope :approved, -> { where(approved: true) }
```

Reply:
BTW what's the concept behind approval here? Is the receiver going to approve them or are admins?